lib/crypto/ed25519/ed25519.go

```diff
@@ -44,6 +44,25 @@ type PublicKey ed25519.PublicKey
 // PublicKeyBytes is an encoded ed25519 public key
 type PublicKeyBytes [PublicKeyLength]byte
 
+// VerifySignature verifies a signature given a public key and a message
+func VerifySignature(publicKey, signature, message []byte) (bool, error) {
+	pubKey, err := NewPublicKey(publicKey)
+	if err != nil {
+		return false, fmt.Errorf("failed to fetch ed25519 public key: %w", err)
+	}
+	ok, err := pubKey.Verify(message, signature)
+	if err != nil {
+		return false, fmt.Errorf("ed25519: %w: for message 0x%x, signature 0x%x and public key 0x%x",
+			crypto.ErrSignatureVerificationFailed, message, signature, publicKey)
+	}
+
+	if !ok {
+		return false, nil
+	}
+
+	return true, nil
```

Comment:
Please change the signatures of all `VerifySignature` functions to only return an error. `ok` is not needed here since we return `crypto.ErrSignatureVerificationFailed` if the verification failed.